src/app/shared/services/app-config/app-config.service.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable } from "@angular/core";
+import { getDefaultAppConfig, IAppConfig, IAppConfigOverride } from "data-models";
+import { BehaviorSubject } from "rxjs";
+import { environment } from "src/environments/environment";
+import { deepMergeObjects } from "../../utils";
+import { SkinService } from "../skin/skin.service";
+
+@Injectable({
+  providedIn: "root",
+})
+export class AppConfigService {
+  appConfigDeploymentOverrides: IAppConfigOverride =
+    (environment.deploymentConfig as any).app_config || {};
+  appConfigSkinOverrides: IAppConfigOverride;
+  /** List of constants provided by data-models combined with deployment-specific overrides and skin-specific overrides */
+  appConfig$ = new BehaviorSubject<IAppConfig | {}>({});
+  /** Static value to be read by methods that are not responsive to changes
+   * TODO: migrate all consumers of this value to be responsive to appConfig$ */
+  APP_CONFIG: IAppConfig;
+
+  constructor(private skinService: SkinService) {}
+
+  init() {
+    this.updateAppConfig();
+    this.skinService.activeSkin$.subscribe((activeSkin) => {
```

Comment:
nit(blocking)
In terms of hierarchy I think the appConfig service should sit above/independent of the skin service, so instead of listening for changes the skin service should call the appConfig service to update config instead